core/services/relay/evm/functions/config_poller.go

```diff
@@ -0,0 +1,192 @@
+package functions
+
+import (
+	"context"
+	"database/sql"
+	"encoding/binary"
+
+	"github.com/ethereum/go-ethereum/accounts/abi"
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/pkg/errors"
+	"github.com/smartcontractkit/libocr/gethwrappers2/ocr2aggregator"
+	ocrtypes "github.com/smartcontractkit/libocr/offchainreporting2/types"
+
+	"github.com/smartcontractkit/chainlink/v2/core/chains/evm/logpoller"
+	"github.com/smartcontractkit/chainlink/v2/core/logger"
+	"github.com/smartcontractkit/chainlink/v2/core/services/pg"
+)
+
+var _ ConfigPoller = &configPoller{}
+
+type ConfigPoller interface {
+	ocrtypes.ContractConfigTracker
+
+	Start()
+	Close() error
+	Replay(ctx context.Context, fromBlock int64) error
+}
+
+type FunctionsPluginType int
+
+const (
+	FunctionsPlugin FunctionsPluginType = iota
+	ThresholdPlugin
+	S4Plugin
+)
+
+type configPoller struct {
+	lggr               logger.Logger
+	filterName         string
+	destChainLogPoller logpoller.LogPoller
+	addr               common.Address
+	pluginType         FunctionsPluginType
+}
+
+// ConfigSet Common to all OCR2 evm based contracts: https://github.com/smartcontractkit/libocr/blob/master/contract2/dev/OCR2Abstract.sol
+var ConfigSet common.Hash
+
+var defaultABI abi.ABI
+
+const configSetEventName = "ConfigSet"
+
+func init() {
+	var err error
+	abiPointer, err := ocr2aggregator.OCR2AggregatorMetaData.GetAbi()
+	if err != nil {
+		panic(err)
+	}
+	defaultABI = *abiPointer
+	ConfigSet = defaultABI.Events[configSetEventName].ID
+}
+
+func unpackLogData(d []byte) (*ocr2aggregator.OCR2AggregatorConfigSet, error) {
+	unpacked := new(ocr2aggregator.OCR2AggregatorConfigSet)
+	err := defaultABI.UnpackIntoInterface(unpacked, configSetEventName, d)
+	if err != nil {
+		return nil, errors.Wrap(err, "failed to unpack log data")
+	}
+	return unpacked, nil
+}
+
+func configFromLog(logData []byte, pluginType FunctionsPluginType) (ocrtypes.ContractConfig, error) {
```

Comment:
Consider the same suggestion that I made below for Digester, with a go-playground example.

If that's not easily doable for the Poller then maybe call the other configFromLog() function (make it public) and then modify the ConfigDigest field afterwards? That way you wouldn't need your own copy of unpackLogData() at all.

Reply:
Since the original `configFromlog` is in the `evm` package, I cannot use it without moving it, else I get a circular import.  While I don't love the duplicated code, I think the current approach might still be the best option for readability.